app/src/main/java/com/example/dogadoption/viewmodels/DogViewModel.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.dogadoption.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.dogadoption.repository.DogRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class DogViewModel @Inject constructor(private val dogRepository: DogRepository) : ViewModel() {
+    private val _dogBreeds = MutableLiveData<List<String>>()
+    val dogBreeds: LiveData<List<String>> get() = _dogBreeds
+
+    fun fetchDogBreeds() {
+        viewModelScope.launch {
```

Comment:
in the constructor you should pass the `dispatcher: CoroutineDispatcher` and use it in launch `viewModelScope.launch(dispatcher) {..}`